core/io/zip_io.cpp

```diff
@@ -87,9 +87,7 @@ long zipio_seek(voidpf opaque, voidpf stream, uLong offset, int origin) {
 int zipio_close(voidpf opaque, voidpf stream) {
 	FileAccess *&f = *(FileAccess **)opaque;
 	if (f) {
-		f->close();
 		memdelete(f);
-		f = nullptr;
```

Comment:
Needs review too as I'm not sure about what `FileAccess *&f` is exactly here. If this was somewhat able to set `opaque` to a `nullptr` this might need to be re-added.